compiler/rustc_hir_analysis/src/impl_wf_check.rs

```diff
@@ -80,7 +85,10 @@ fn enforce_impl_params_are_constrained(tcx: TyCtxt<'_>, impl_def_id: LocalDefId)
                 "potentially unconstrained type parameters weren't evaluated: {impl_self_ty:?}",
             ),
         );
-        return;
+        // This is super fishy, but our current `rustc_hir_analysis::check_crate` pipeline depends on
+        // `type_of` having been called much earlier, and thus this value being read from cache.
+        // Compilation must continue in order for other important diagnostics to keep showing up.
+        return Ok(());
```

Comment:
The new errors in the tests happen because of this hack.

* If we `track_errors` the `type_of` call above, we'd get zero diagnostics changes.
* If we use `error_reported()?` on the type returned from `type_of`, we lose lots of useful diagnostics
* If we ignore the `Result` returned from `check_mod_impl_wf`, we get lots of new diagnostics, sometimes not very useful.

We can change this easily in the future, but I'd rather write separate PRs to compare, and not do it in this PR.